### DISCUSSION TITLE
Why Zed automatically closes double quotations but not single quotations?

### DISCUSSION THREAD

**Discussion by albas99** [zed-industries/zed#6632] [Feature Requests]
I don't know whether this is due to my lsp(pyright) or from zed, but it seems like when I'm using double quotations the editor automatically add a closing quote for me but when I using a single quote, it refuses to add the closing quote for me. Can someone help explain to me why's that? Or whether it's from my Lsp?

**Comment by iamnbutler** [comment]
Is this specifically for Python, or you are observing this with other languages as well?

**Reply by albas99** [reply]
Hey @iamnbutler, sorry for the late reply, your message got buried in my notifications and I'm just seeing it. So from what I've tested I don't seem to be getting the issue in Javascript, however I do get it in python and html. Here's a screen recording testing it on JS, Python and HTML: https://www.loom.com/share/ab20b2db948144e2a0597ce1cff49401?sid=067d8d5c-d6a5-4d7d-9ed0-6d36f0e9f582

**Comment by gldtn** [comment]
I don't get single quote closing either, tried it in HTML/PHP/JSON. I know this thread is a bit old, but I am using:

Zed: v0.111.3 (preview)
OS: macOS 14.0.0
Architecture: aarch64

**Comment by a1go3** [comment]
I have this problem too. Zed, does not close single quotes in Python :(
Are there any options to fix this?

**Comment by tibitTW** [comment]
I also met this issue in Python

Zed: v0.141.3 (Zed)
OS: macOS 14.5.0
Memory: 8 GiB
Architecture: aarch64

**Comment by frank-laemmer** [comment]
Seems to be related (duplicate?):
https://github.com/zed-industries/zed/discussions/7951

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
